doc/stdlib/base.rst

```diff
@@ -1218,11 +1218,13 @@ Errors
 
    An error occurred when running a module's ``__init__`` function. The actual error thrown is available in the ``.error`` field.
 
-.. function:: retry(f, [condition]; n=3; max_delay=10) -> Function
+.. function:: retry(f, [retry_on]; n=1, max_delay=0.05) -> Function
 
    .. Docstring generated from Julia source
 
-   Returns a lambda that retries function ``f`` up to ``n`` times in the event of an exception. If ``condition`` is a ``Type`` then retry only for exceptions of that type. If ``condition`` is a function ``cond(::Exception) -> Bool`` then retry only if it is true.
+   Returns a lambda that retries function ``f`` up to ``n`` times in the event of an exception. If ``retry_on`` is a ``Type`` then retry only for exceptions of that type. If ``retry_on`` is a function ``test_error(::Exception) -> Bool`` then retry only if it is true.
+
+   The first retry happens after a gap of 50 milliseconds or ``max_delay``\ , whichever is lower. Subsequently, the delays between retries are exponentially increased with a random factor upto ``max_delay``\ .
```

Comment:
It seems odd to me to describe the exponential back off but have the default `max_delay` set so that there is no exponential back-off.
Would you consider leaving `max_delay` at 10 in this PR so that it can be discussed further before it is changed.
After all now that default `n` is `1` the `max_delay` has not effect anyway unless the user changes `n`.